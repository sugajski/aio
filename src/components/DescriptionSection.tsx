import React, { FC } from "react";
import { Text } from "react-native";

import DescriptionSectionStyles from "./styles/DescriptionSectionStyles";

interface IDescriptionSectionProps {
  author: string;
  description: string;
}

const DescriptionSection: FC<IDescriptionSectionProps> = ({
  author,
  description,
}) => (
  <Text>
    <Text style={DescriptionSectionStyles.author}>{author} </Text>
    <Text style={DescriptionSectionStyles.description}>{description}</Text>
  </Text>
);

export default DescriptionSection;
