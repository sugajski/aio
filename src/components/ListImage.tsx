import React, { FC } from "react";
import { TouchableOpacity } from "react-native";
import { Image as CachedImage } from "react-native-expo-image-cache";

import ListImageStyles from "./styles/ListImageStyles";

interface IListImageProps {
  uri: string;
  onPress: () => void;
}

const ListImage: FC<IListImageProps> = ({ uri, onPress }) => (
  <TouchableOpacity onPress={onPress}>
    <CachedImage uri={uri} style={ListImageStyles.image} />
  </TouchableOpacity>
);

export default ListImage;
