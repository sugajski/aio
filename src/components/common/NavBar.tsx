import React, { useEffect, FC } from "react";
import { View, Image, TouchableOpacity, BackHandler, Text } from "react-native";
import { useHistory } from "react-router-native";

import { Images } from "../../themes";
import NavBarStyles from "./styles/NavBarStyles";

interface INavBarProps {
  title?: string;
}

const NavBar: FC<INavBarProps> = ({ title = null }) => {
  const router = useHistory();

  useEffect(() => {
    BackHandler.addEventListener("hardwareBackPress", handleBackPressed);
    return () => {
      BackHandler.removeEventListener("hardwareBackPress", handleBackPressed);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleBackPressed = () => {
    router.goBack();
    return true;
  };

  return (
    <View style={NavBarStyles.container}>
      {title && <Text style={NavBarStyles.title}>{title}</Text>}
      <TouchableOpacity onPress={handleBackPressed}>
        <View style={NavBarStyles.backButtonContainer}>
          <Image
            source={Images.back}
            style={NavBarStyles.backButtonIcon}
            resizeMode="stretch"
          />
        </View>
      </TouchableOpacity>
    </View>
  );
};

export default NavBar;
