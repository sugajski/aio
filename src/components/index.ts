import DescriptionSection from './DescriptionSection';
import ListImage from './ListImage';

export { DescriptionSection, ListImage };
