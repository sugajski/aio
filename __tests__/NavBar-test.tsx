import React from "react";
import renderer from "react-test-renderer";
import { SafeAreaProvider } from "react-native-safe-area-context";

import { NavBar } from "../src/components/common";

test("NavBar renders correctly", () => {
  const tree = renderer
    .create(
      <SafeAreaProvider
        initialSafeAreaInsets={{ top: 0, left: 0, right: 0, bottom: 0 }}
      >
        <NavBar />
      </SafeAreaProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
