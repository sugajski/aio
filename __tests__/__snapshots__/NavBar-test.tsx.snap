// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flexDirection": "row",
        "height": 56,
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 56,
            "justifyContent": "center",
            "width": 56,
          }
        }
      >
        <Image
          resizeMode="stretch"
          source={1}
          style={
            Object {
              "height": 18,
              "width": 14,
            }
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
